src/smpi/smpi_pmpi.cpp

```diff
@@ -907,16 +910,92 @@ int PMPI_Ssend_init(void* buf, int count, MPI_Datatype datatype, int dst, int ta
   return retval;
 }
 
-int PMPI_Start(MPI_Request * request)
+int persistent_send(MPI_Request *request)
 {
   int retval = 0;
+  MPI_Request req = *request;
 
-  smpi_bench_end();
+  //smpi_bench_end();
   if (request == nullptr || *request == MPI_REQUEST_NULL) {
     retval = MPI_ERR_REQUEST;
   } else {
+
+    int rank = req->comm != MPI_COMM_NULL ? smpi_process_index() : -1;
```

Comment:
This is function is based on PMPI_Isend (actually, most of the code was copied from it). This test is present in that function too.